File: app/src/main/java/com/omvp/app/service/AppFirebaseMessagingService.java
package com.omvp.app.service;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.omvp.app.util.DisposableManager;

import javax.inject.Inject;
import javax.inject.Named;

import dagger.android.AndroidInjection;
import timber.log.Timber;

import static com.omvp.app.base.BaseServiceModule.DISPOSABLE_SERVICE_MANAGER;

public class AppFirebaseMessagingService extends FirebaseMessagingService {

    @Inject
    @Named(DISPOSABLE_SERVICE_MANAGER)
    protected DisposableManager mDisposableManager;

    @Override
    public void onCreate() {
        AndroidInjection.inject(this);
        super.onCreate();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        mDisposableManager.dispose();
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        Timber.d("onMessageReceived from: %s", remoteMessage.getFrom());
        // Check if message contains a data payload.
        if (remoteMessage.getData().size() > 0) {
            Timber.d("Message data payload: %s", remoteMessage.getData());
        }
        // Check if message contains a notification payload.
        if (remoteMessage.getNotification() != null) {
            Timber.d("Message Notification Body: %s", remoteMessage.getNotification().getBody());
        }
    }
}

File: app/src/main/java/com/omvp/app/service/AppFirebaseMessagingService.kt
package com.omvp.app.service

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.omvp.app.base.BaseServiceModule.Companion.DISPOSABLE_SERVICE_MANAGER
import com.omvp.app.util.DisposableManager
import dagger.android.AndroidInjection
import timber.log.Timber
import javax.inject.Inject
import javax.inject.Named

class AppFirebaseMessagingService : FirebaseMessagingService() {

    @Inject
    @field:Named(DISPOSABLE_SERVICE_MANAGER)
    lateinit var mDisposableManager: DisposableManager

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    override fun onDestroy() {
        super.onDestroy()

        mDisposableManager.dispose()
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {
        Timber.d("onMessageReceived from: %s", remoteMessage!!.from)
        // Check if message contains a data payload.
        if (remoteMessage.data.isNotEmpty()) {
            Timber.d("Message data payload: %s", remoteMessage.data)
        }
        // Check if message contains a notification payload.
        if (remoteMessage.notification != null) {
            Timber.d("Message Notification Body: %s", remoteMessage.notification!!.body)
        }
    }
}
